### DISCUSSION TITLE
Resource inheritance in a subview

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#11976] [Q&A] [answered]
In the project attached i declare a converter with `x:Key="tc"`

```xaml 
  <Window.Resources>
    <local:TestValueConverter x:Key="tc"/>
    <sys:Decimal x:Key="Value">12.3</sys:Decimal>
  </Window.Resources>
```

and tried calling the converter from a `UserControl` called TestView inside a DataTemplate.

this is code is definition of usercontrol

```xaml
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="Sandbox.TestView"
             xmlns:sys="using:System"
             x:DataType="sys:String">
  <TextBox x:Name="textBlock" Text="{Binding ., Converter={StaticResource tc} }"/>
</UserControl>
```
this is the code of how to use it

```xaml
    <ItemsControl ItemsSource="{Binding ObservableCollectionProperty}">
      <ItemsControl.ItemTemplate>
        <DataTemplate DataType="{x:Type sys:String}">
          <local:TestView/>
        </DataTemplate>
      </ItemsControl.ItemTemplate>
    </ItemsControl>
```

When I run it throw `Static resource 'tc' not found.`, is this an expected behavior?

in order not to throw the exception I'm forced to move it to App.axaml.
but it is not functional to my real application because I need to pass a value present in the main view to the converter.

[Sandbox.zip](https://github.com/AvaloniaUI/Avalonia/files/11916988/Sandbox.zip)

**Comment by maxkatz6** [comment, marked as answer]
It is expected, as you use static resource from the control before it is attached on any parent. 
Basically:
```csharp
var testView = new TestView(); // textBlock is created at this point, and static resource is resolved too
window.Content = testView; // but only at this point "tc" resource will be available
```

DynamicResource doesn't have this problem, as it will return null initially, and then re-evaluate when view is attached to the parent.
And it also works with Application resources, as it's always a global fallback.

**Reply by workgroupengineering** [reply]
Thank!!!

**Reply by workgroupengineering** [reply]
Sorry to bother you again, I tried to replace StaticResource with DynamicResource, now the compiler throws the following error.

`Unable to find suitable setter or adder for property Converter of type Avalonia.Markup:Avalonia.Data.BindingBase for argument Avalonia.Base:Avalonia.Data.IBinding, available setter parameter lists are: Avalonia.Data.Converters.IValueConverter Line 9, position 31.	Sandbox	C:\GitHub\Avalonia\samples\Sandbox\TestView.axaml	9`

Do you have any idea how to fix?

**Reply by timunie** [reply]
Probably in TestValueConverter add a static instance: 

```cs
public static TestValueConverter Instance {get; } = new TestValueConverter();
```

and usage is then: 
```xml
<TextBox x:Name="textBlock" Text="{Binding ., Converter={x:static conv:TestValueConverter.Instance} }"/>
```

**Reply by workgroupengineering** [reply]
Thanks for your suggestion.

Actually the actual problem is more complex than the example I have attached.

The converter I implemented also inherits from `AvaloniaObject` and had a property binding to the `MainViewModel`. A kind of WPF `BindingProxy`. In the end I opted for `IMultiValueConverter`
